cmd/gapit/export_replay.go

```diff
@@ -98,5 +114,155 @@ func (verb *exportReplayVerb) Run(ctx context.Context, flags flag.FlagSet) error
 	if err := client.ExportReplay(ctx, capturePath, device, verb.Out, opts); err != nil {
 		return log.Err(ctx, err, "Failed to export replay")
 	}
+
+	if verb.Apk != "" {
+		// Create stand-alone APK
+		replayAPK := filepath.Base(verb.Apk)
+		replayPackage := strings.TrimSuffix(replayAPK, ".apk")
+		log.I(ctx, "Create replay apk: %s with package name %s", replayAPK, replayPackage)
+
+		boxedCapture, err := client.Get(ctx, capturePath.Path(), nil)
+		if err != nil {
+			return log.Err(ctx, err, "Failed to load the capture")
+		}
+		capture := boxedCapture.(*service.Capture)
+
+		refAPKPath, err := layout.GapidApk(ctx, capture.ABI)
+		if err != nil {
+			return err
+		}
+
+		// Extract reference APK
+		if err := execCommand(ctx, "unzip", "-o", refAPKPath.String()); err != nil {
+			return err
+		}
+
+		// Create new manifest
+		f, err := os.Create("NewManifest.xml")
+		if err != nil {
+			return err
+		}
+		t, err := template.New("manifest").Parse(manifest())
+		if err != nil {
+			return err
+		}
+		type ManifestEntries struct {
+			Package string
+		}
+		var entries = ManifestEntries{replayPackage}
+		t.Execute(f, entries)
+		f.Close()
+
+		// Re-assemble new APK with the new manifest
+		if err := execCommand(ctx, "aapt2", "link", "-o", verb.Apk, "--manifest", "NewManifest.xml", "-I", os.ExpandEnv("${ANDROID_SDK_HOME}/platforms/android-26/android.jar")); err != nil {
+			return err
+		}
+
+		// Add replay assets, uncompressed
+		assets, err := ioutil.ReadDir("assets/replay_export/")
+		if err != nil {
+			return err
+		}
+		for _, a := range assets {
+			// Arguments ("-0", "") ensure the asset will not be compressed
+			if err := execCommand(ctx, "aapt", "add", "-0", "", replayAPK, "assets/replay_export/"+a.Name()); err != nil {
+				return err
+			}
+		}
+
+		// Add the replay libraries
+		// TODO(#2677): get rid of this workaround once ABI naming is sanitized
+		abi := capture.ABI.Name
+		if abi == "armeabi" {
+			abi = "armeabi-v7a"
+		}
+		if err := execCommand(ctx, "zip", "-u", replayAPK, "lib/"+abi+"/libgapir.so", "lib/"+abi+"/libVkLayer_VirtualSwapchain.so"); err != nil {
+			return err
+		}
+
+		// Add the compiled Java code
+		if err := execCommand(ctx, "zip", "-uj", replayAPK, "classes.dex"); err != nil {
+			return err
+		}
+
+		// Sign the new APK
+		if err := execCommand(ctx, "apksigner", "sign", "--ks", os.ExpandEnv("${HOME}/.android/debug.keystore"), "--ks-pass", "pass:android", replayAPK); err != nil {
```

Comment:
I dont think this env will work on Windows.
We should probably set up a dummy keystore if none exist.

Reply:
I think we need to run zipalign on the zi pfile before apksigner as well?